File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.cpp
class Solution {
public:
    bool vowels(char ch){
        if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || 
        ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U'){
            return true;
        }
        return false;
    }

    string sortVowels(string s) {
        string v = "";
        for(int i = 0;i < s.size();i++){
            if(vowels(s[i])){
                v.push_back(s[i]);
            }
        }      
        sort(v.begin(),v.end());
        int idx = 0;
        string ans = "";
        for(int i = 0;i < s.size();i++){
            if(vowels(s[i])){
                ans.push_back(v[idx]);
                idx++;
            }
            else{
                ans.push_back(s[i]);
            }
        }
        return ans;

    }
};

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public boolean isvowel(char ch){
        if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' ||
        ch == 'A' || ch == 'E' || ch == 'O' ||ch == 'I' ||ch == 'U'){
            return true;
        }
        return false;
    }


    public String sortVowels(String s) {
        StringBuilder v = new StringBuilder();
        StringBuilder ans = new StringBuilder();

        for(int i = 0;i < s.length();i++){
            if(isvowel(s.charAt(i))){
                v.append(s.charAt(i));
            }
        }
        char[] vo = v.toString().toCharArray();
        Arrays.sort(vo);
        int idx = 0;
        for(int i = 0;i < s.length();i++){
            if(isvowel(s.charAt(i))){
                ans.append(vo[idx]);
                idx++;
            }
            else{
                ans.append(s.charAt(i));
            }
        }
        return ans.toString();
        
    }
}